### PR TITLE
Some run configurations refactorings

### PR DESCRIPTION
**Pull request by mkaput**
I am tinkering (slowly) on test runner stuff, but debugger related commits brought some terrible merge conflicts 😭 

Here are some cleanups (mostly moved some code, nothing especially new) I did and managed to extract from my playground.

### CODE REVIEW

File: src/main/kotlin/org/rust/cargo/runconfig/filters/RegexpFileLinkFilter.kt [hunk 1]
```diff
@@ -45,7 +45,7 @@ open class RegexpFileLinkFilter(
         val columnNumber = match.groups[3]?.let { zeroBasedNumber(it.value) } ?: 0
 
         val lineStart = entireLength - line.length
-        return Filter.Result(
+        return com.intellij.execution.filters.Filter.Result(
```
Comment (matklad) [inline src/main/kotlin/org/rust/cargo/runconfig/filters/RegexpFileLinkFilter.kt]: Let's import this thing?

File: src/main/kotlin/org/rust/cargo/runconfig/filters/RsBacktraceFilter.kt [hunk 2]
```diff
@@ -66,9 +66,9 @@ private class RsBacktraceItemFilter(
         }
 
         // Dim the hashcode
-        resultItems.add(Filter.ResultItem(funcEnd, funcEnd + funcHash.length, null, DIMMED_TEXT))
+        resultItems.add(com.intellij.execution.filters.Filter.ResultItem(funcEnd, funcEnd + funcHash.length, null, DIMMED_TEXT))
 
-        return Filter.Result(resultItems)
+        return com.intellij.execution.filters.Filter.Result(resultItems)
```
Comment (matklad) [inline src/main/kotlin/org/rust/cargo/runconfig/filters/RsBacktraceFilter.kt]: Same here

**Comment by matklad** [comment]
Yeah, I not happy with how `RunConfigurations` work currently either!

File: src/main/kotlin/org/rust/cargo/runconfig/filters/RegexpFileLinkFilter.kt (hunk 1 above)
Comment (mkaput) [inline src/main/kotlin/org/rust/cargo/runconfig/filters/RegexpFileLinkFilter.kt]: Ugh, Kotlin's `Move refactoring` did some strange things I see

File: src/main/kotlin/org/rust/cargo/runconfig/command/CargoExecutableRunLineMarkerContributor.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.util.elementType
+
+class CargoExecutableRunLineMarkerContributor : RunLineMarkerContributor() {
+    override fun getInfo(element: PsiElement): Info? {
+        if (element.elementType != IDENTIFIER) return null
+        val fn = element.parent as? RsFunction ?: return null
+        if (!CargoExecutableRunConfigurationProducer.isMainFunction(fn)) return null
+
+        val actions = ExecutorAction.getActions(0)
```
Comment (mkaput) [inline src/main/kotlin/org/rust/cargo/runconfig/command/CargoExecutableRunLineMarkerContributor.kt]: BTW: Is it possible to filter available executors? (overally, not here, though filtering by `toString()` or by using reflection to extract private fields isn't good idea either)

User gets spammed with `Debug` and `Run with Coverage` actions which are unavailable as for now.

Comment (matklad) [inline src/main/kotlin/org/rust/cargo/runconfig/command/CargoExecutableRunLineMarkerContributor.kt]: https://github.com/intellij-rust/intellij-rust/commit/736a4d8d3515277fbb113b9a47e6214a95b2d342

though it does not work in this case for some reason.

Comment (matklad) [inline src/main/kotlin/org/rust/cargo/runconfig/command/CargoExecutableRunLineMarkerContributor.kt]: I'll merge it and do cleanup myself!

**Comment by matklad** [comment]
Thanks @mkaput !

File: src/main/kotlin/org/rust/cargo/runconfig/command/CargoExecutableRunLineMarkerContributor.kt (hunk 3 above)
Comment (mkaput) [inline src/main/kotlin/org/rust/cargo/runconfig/command/CargoExecutableRunLineMarkerContributor.kt]: Thanks! :)

**Pull request merged**